homeassistant/helpers/typing.py

```diff
@@ -26,6 +26,16 @@ class UndefinedType(Enum):
 
 UNDEFINED = UndefinedType._singleton  # pylint: disable=protected-access
 
+
+class ConversionUtility(Protocol):
+    """Define the format of a conversion utility."""
+
+    VALID_UNITS: tuple[str, ...]
+
+    def convert(self, value: float, from_unit: str, to_unit: str) -> float:
+        """Convert one unit of measurement to another."""
+
+
```

Comment:
I couldn't find a better place to put this `Protocol`
I think it's a typing helper, but if someone proposes a better place for it I'd be happy to oblige